shared/js/background/redirect.es6.js

```diff
@@ -10,6 +10,23 @@ var debugRequest = false
 
 trackers.loadLists()
 
+var ports = []
+function connected(p) {
+    ports[p.sender.frameId] = p;
+	console.log("p value", p.sender.tab.id);
+    let requestTab = tabManager.get({tabId: p.sender.tab.id});
+    console.log("thisTab", requestTab);
+    p.onMessage.addListener(function(m) {
+        if (m.frame === 'main') {
+            p.postMessage({blockedRequests: requestTab.framesBlocked});
+        } else if (m.frame === 'sub') {
+            p.postMessage({blockedRequests: requestTab.scriptsAndFramesBlocked});
+        }
+    })
+}
+
+chrome.runtime.onConnect.addListener(connected);
```

Comment:
Can we move this to `chrome-events`?

Reply:
@andrey-p definitely. I still need to refactor and organize the code quite a bit, but just wanted to get a working version up here to get more eyes on it.

Reply:
Sorry, I'm jumping the gun a bit :gun: :man_cartwheeling:

Reply:
No worries at all! I'm just happy you're looking at it already.